flexmeasures/data/models/reporting/pandas_reporter.py

```diff
@@ -88,48 +93,89 @@ def _compute_report(self, **kwargs) -> tb.BeliefsDataFrame:
             resolution = self.sensor.event_resolution
 
         # fetch sensor data
-        self.fetch_data(start, end, input_variables, resolution, belief_time)
+        self.fetch_data(start, end, input, resolution, belief_time)
 
         if belief_time is None:
             belief_time = server_now()
 
         # apply pandas transformations to the dataframes in `self.data`
         self._apply_transformations()
 
-        final_output = self.data[self._config.get("final_df_output")]
-
-        if isinstance(final_output, tb.BeliefsDataFrame):
-
-            # filing the missing indexes with default values:
-            # belief_time=belief_time, cummulative_probability=0.5, source=data_source
-            if "belief_time" not in final_output.index.names:
-                final_output["belief_time"] = [belief_time] * len(final_output)
-                final_output = final_output.set_index("belief_time", append=True)
-
-            if "cumulative_probability" not in final_output.index.names:
-                final_output["cumulative_probability"] = [0.5] * len(final_output)
-                final_output = final_output.set_index(
-                    "cumulative_probability", append=True
-                )
-
-            if "source" not in final_output.index.names:
-                final_output["source"] = [self.data_source] * len(final_output)
-                final_output = final_output.set_index("source", append=True)
-
-            final_output = final_output.reorder_levels(
-                tb.BeliefsDataFrame().index.names
-            )
+        output = kwargs.get("output", [])
 
-        elif isinstance(final_output, tb.BeliefsSeries):
-            final_output = final_output.to_frame("event_value")
-            final_output["belief_time"] = belief_time
-            final_output["cumulative_probability"] = 0.5
-            final_output["source"] = self.data_source
-            final_output = final_output.set_index(
-                ["belief_time", "source", "cumulative_probability"], append=True
+        if len(output) == 0 and output_sensor is None:
+            raise ValueError(
+                "No output sensor defined. At least define an output sensor in the `sensor` or `output` fields in `parameters`."
```

Comment:
Perhaps this message can be copied to the one where I commented that the sentence doesn't run well.

Reply:
`output_sensor` now it's gone, so this if doesn't make sense anymore